wasm/validator/derive/src/default.rs

```diff
@@ -0,0 +1,218 @@
+        "visit_instruction(&InstructionBox)",
+        "visit_expression<V>(&EvaluatesTo<V>)",
+        "visit_sequence(&SequenceBox)",
+        "visit_if(&Conditional)",
+        "visit_pair(&Pair)",
+        "visit_unregister_peer(Unregister<Peer>)",
+        "visit_unregister_domain(Unregister<Domain>)",
+        "visit_set_domain_key_value(SetKeyValue<Domain>)",
+        "visit_remove_domain_key_value(RemoveKeyValue<Domain>)",
+        "visit_unregister_account(Unregister<Account>)",
+        "visit_mint_account_public_key(Mint<Account, PublicKey>)",
+        "visit_burn_account_public_key(Burn<Account, PublicKey>)",
+        "visit_mint_account_signature_check_condition(Mint<Account, SignatureCheckCondition>)",
+        "visit_set_account_key_value(SetKeyValue<Account>)",
+        "visit_remove_account_key_value(RemoveKeyValue<Account>)",
+        "visit_register_asset(Register<Asset>)",
+        "visit_unregister_asset(Unregister<Asset>)",
+        "visit_mint_asset(Mint<Asset, NumericValue>)",
+        "visit_burn_asset(Burn<Asset, NumericValue>)",
+        "visit_transfer_asset(Transfer<Asset, NumericValue, Account>)",
+        "visit_set_asset_key_value(SetKeyValue<Asset>)",
+        "visit_remove_asset_key_value(RemoveKeyValue<Asset>)",
+        "visit_unregister_asset_definition(Unregister<AssetDefinition>)",
+        "visit_transfer_asset_definition(Transfer<Account, AssetDefinition, Account>)",
+        "visit_set_asset_definition_key_value(SetKeyValue<AssetDefinition>)",
+        "visit_remove_asset_definition_key_value(RemoveKeyValue<AssetDefinition>)",
+        "visit_grant_account_permission(Grant<Account, PermissionToken>)",
+        "visit_revoke_account_permission(Revoke<Account, PermissionToken>)",
+        "visit_register_role(Register<Role>)",
+        "visit_unregister_role(Unregister<Role>)",
+        "visit_grant_account_role(Grant<Account, RoleId>)",
+        "visit_revoke_account_role(Revoke<Account, RoleId>)",
+        "visit_unregister_trigger(Unregister<Trigger<TriggeringFilterBox, Executable>>)",
+        "visit_mint_trigger_repetitions(Mint<Trigger<TriggeringFilterBox, Executable>, u32>)",
+        "visit_execute_trigger(ExecuteTrigger)",
+        "visit_set_parameter(SetParameter)",
+        "visit_new_parameter(NewParameter)",
+        "visit_upgrade_validator(Upgrade<iroha_validator::data_model::validator::Validator>)",
+    ]
+    .into_iter()
+    .map(syn::parse_str)
+    .map(|item| item.expect("Function names and operation signatures should be valid"))
+    .collect();
+
+    let idents_to_exclude = custom_items.0.as_ref().map_or_else(
+        || vec![],
+        |impl_| {
+            let items = &impl_.items;
+            items
+                .iter()
+                .map(|item| match item {
+                    syn::ImplItem::Method(method) => &method.sig.ident,
+                    _ => panic!("Only methods are expected in `Visit` impl blocks"),
```

Comment:
don't use panics. @DCNick3 what are we using instead of `proc-macro-error` now?

Reply:
`manyhow`, and, where applicable, `iroha_macro_utils::Emitter` (though it's not yet merged, see #3882)

Reply:
Will do that. I suppose it's better not to touch refactoring the remainder of the `validator_derive` crate to keep things granular?